File: 351003/Mashevskiy/Lab3/publisher/src/main/java/com/distcomp/publisher/service/CreatorService.java
package com.distcomp.publisher.service;

import com.distcomp.publisher.dto.CreatorRequestDTO;
import com.distcomp.publisher.dto.CreatorResponseDTO;
import com.distcomp.publisher.model.Creator;
import com.distcomp.publisher.repository.CreatorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class CreatorService {

    @Autowired
    private CreatorRepository creatorRepository;

    public List<CreatorResponseDTO> getAll() {
        return creatorRepository.findAll().stream()
                .map(this::toResponse)
                .collect(Collectors.toList());
    }

    public CreatorResponseDTO getById(Long id) {
        Creator creator = creatorRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Creator not found"));
        return toResponse(creator);
    }

    public CreatorResponseDTO create(CreatorRequestDTO dto) {
        Creator creator = new Creator();
        creator.setLogin(dto.getLogin());
        creator.setPassword(dto.getPassword());
        creator.setFirstname(dto.getFirstname());
        creator.setLastname(dto.getLastname());
        return toResponse(creatorRepository.save(creator));
    }

    public CreatorResponseDTO update(Long id, CreatorRequestDTO dto) {
        Creator creator = creatorRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Creator not found"));
        creator.setLogin(dto.getLogin());
        creator.setPassword(dto.getPassword());
        creator.setFirstname(dto.getFirstname());
        creator.setLastname(dto.getLastname());
        return toResponse(creatorRepository.save(creator));
    }

    public void delete(Long id) {
        creatorRepository.deleteById(id);
    }

    private CreatorResponseDTO toResponse(Creator creator) {
        return new CreatorResponseDTO(creator.getId(), creator.getLogin(),
                creator.getFirstname(), creator.getLastname());
    }
}

File: 351004/Leshok/lab3/publisher/src/services/CreatorService.js
const { Creator } = require('../models');
const { CreatorResponseDto } = require('../dtos/CreatorDto');

class CreatorService {
  async create(dto) {
    const { login, password, firstname, lastname } = dto;
    const creator = await Creator.create({ login, password, firstname, lastname });
    return new CreatorResponseDto(creator.id, creator.login, creator.password, creator.firstname, creator.lastname);
  }

  async update(id, dto) {
    const creator = await Creator.findByPk(id);
    if (!creator) throw { status: 404, code: '40401', message: 'Creator not found' };
    const { login, password, firstname, lastname } = dto;
    await creator.update({ login, password, firstname, lastname });
    return new CreatorResponseDto(creator.id, creator.login, creator.password, creator.firstname, creator.lastname);
  }

  async getById(id) {
    const creator = await Creator.findByPk(id);
    if (!creator) throw { status: 404, code: '40401', message: 'Creator not found' };
    return new CreatorResponseDto(creator.id, creator.login, creator.password, creator.firstname, creator.lastname);
  }

  async getAll() {
    const creators = await Creator.findAll();
    return creators.map(c => new CreatorResponseDto(c.id, c.login, c.password, c.firstname, c.lastname));
  }

  async delete(id) {
    const creator = await Creator.findByPk(id);
    if (!creator) throw { status: 404, code: '40401', message: 'Creator not found' };
    await creator.destroy();
  }
}

module.exports = new CreatorService();